### PR TITLE
Automated tests for the CRUD spec ("find" and "insert_one")

### PR DESCRIPTION
**Pull request by saghmrossi**

### CODE REVIEW

**Comment by saghmrossi** [comment]
@vkarpov15 Looks like the cloning of the "specifications" repo is working.

File: tests/json/eq.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+trait NumEq {
+    fn float_eq(&self, f: f64) -> bool;
+    fn int_eq(&self, i: i64) -> bool;
+}
+
+impl NumEq for Bson {
+    fn float_eq(&self, f: f64) -> bool {
+        match self {
+            &Bson::FloatingPoint(ref ff) => f.approx_eq(ff),
+            &Bson::I32(i) => f == (i as f64),
```
Comment (alabid) [inline tests/json/eq.rs]: should we use `approx_eq` here too?

Comment (saghmrossi) [inline tests/json/eq.rs]: Oops, yeah

File: tests/client/crud_spec/framework.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        assert!(eq::bson_eq(&id, &inserted));
+        check_coll!($db, $c, $o.collection);
+    }};
+}
+
+
+#[macro_export]
+macro_rules! run_suite {
+    ( $f:expr, $c:expr ) => {{
```
Comment (vkarpov15) [inline tests/client/crud_spec/framework.rs]: Let's avoid 1-char variable names, even for macros. Could we write `$file` instead?

File: tests/lib.rs [hunk 3]
```diff
@@ -2,6 +2,9 @@ extern crate bson;
 
 #[macro_use(add_to_doc, doc)]
 extern crate mongodb;
+extern crate rustc_serialize;
+extern crate nalgebra as na;
```
Comment (vkarpov15) [inline tests/lib.rs]: Don't rename packages unless necessary - makes it trickier to dig out what 'na' means.

Comment (saghmrossi) [inline tests/lib.rs]: Okay, I was just following the example in the README:

---
The preferred way to use nalgebra is to import types and traits explicitly, and call free-functions using the na:: prefix:

```
extern crate nalgebra as na;
use na::{Vec3, Rot3, Rotation};

fn main() {
    let     a = Vec3::new(1.0f64, 1.0, 1.0);
    let mut b = Rot3::new(na::zero());

    b.append_rotation_mut(&a);

    assert!(na::approx_eq(&na::rotation(&b), &a));
}
```
---
https://github.com/sebcrozet/nalgebra

File: tests/client/crud_spec/read.rs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+use bson::Bson;
+use json::arguments::Arguments;
+use json::reader::SuiteContainer;
+use json::eq;
+use mongodb::client:: MongoClient;
+use rustc_serialize::json::Json;
+
+#[test]
+fn find() {
+    run_suite!("tests/json/data/specs/source/crud/tests/read/find.json",
```
Comment (alabid) [inline tests/client/crud_spec/read.rs]: nice.

**Pull request merged**